templates/react/packages/client/src/mud/setupNetwork.ts

```diff
@@ -3,12 +3,28 @@
  * (https://viem.sh/docs/getting-started.html).
  * This line imports the functions we need from it.
  */
-import { createPublicClient, fallback, webSocket, http, createWalletClient, Hex, parseEther, ClientConfig } from "viem";
+import {
+  createPublicClient,
+  fallback,
+  webSocket,
+  http,
+  createWalletClient,
+  Hex,
+  parseEther,
+  ClientConfig,
+  getContract,
+} from "viem";
```

Comment:
Now this uses `viem_getContract` instead of `mud_getContract`.

With this custom client setup, there's an opportunity to rethink this template structure (e.g., not doing `getContract` at the setup phase), but I've not explored it here.

Reply:
this is great!

we could consider moving away from the contract instance approach in favor of calling functions with pure viem actions (I think this is recommended by viem team anyway as it saves on bundle sizes)

but happy to defer this to a follow up PR

Reply:
> defer this to a follow up PR

I'll make a follow-up PR for the template improvements!

> consider moving away from the contract instance approach

We can consider this in that PR. If the bundle size difference isn't significant for our app, the contract instance approach might be better for convenience, as we don't need to pass IWorldAbi and worldAddress every time.